lib/improver/tests/nowcasting/convection/handle_precip/test_ApplyPrecip.py

```diff
@@ -0,0 +1,176 @@
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the nowcast.convection.handle_precip.ApplyPrecip plugin."""
+
+import unittest
+from iris.tests import IrisTest
+from iris.cube import Cube
+
+import cf_units
+from improver.nowcasting.convection.handle_precip import ApplyPrecip as Plugin
+from improver.tests.nbhood.nbhood.test_BaseNeighbourhoodProcessing import (
+    set_up_cube, set_up_cube_with_no_realizations)
+from improver.tests.ensemble_calibration.ensemble_calibration.helper_functions\
+    import add_forecast_reference_time_and_forecast_period
+
+
+class Test__repr__(IrisTest):
+
+    """Test the repr method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        result = str(Plugin())
+        msg = ("""<ApplyPrecip:
+ precipitation mapping:
+   upper:  precip probability {precu} => max lightning prob {lprecu}
+   middle: precip probability {precm} => max lightning prob {lprecm}
+   lower:  precip probability {precl} => max lightning prob {lprecl}
+
+   heavy:  prob(precip>7mm/hr)  {pphvy} => min lightning prob {lprobl}
+   intense:prob(precip>35mm/hr) {ppint} => min lightning prob {lprobu}
+>""".format(lprobu=1., lprobl=0.25,
+            precu=0.1, precm=0.05, precl=0.0,
+            lprecu=1., lprecm=0.2, lprecl=0.0067,
+            pphvy=0.4, ppint=0.2))
+        self.assertEqual(result, msg)
+
+
+class Test_process(IrisTest):
+
+    """Test the nowcast convection handle_vii ApplyIce plugin."""
+
+    def setUp(self):
+        """Create a cube with a single non-zero point."""
+        self.fg_cube = add_forecast_reference_time_and_forecast_period(
+            set_up_cube_with_no_realizations(zero_point_indices=[]))
+        self.fg_cube.rename("probability_of_lightning")
+        self.fg_cube.coord('forecast_period').points = [0.]
+        self.precip_cube = (
+            add_forecast_reference_time_and_forecast_period(
+                set_up_cube(num_realization_points=3), fp_point=0.0))
+        threshold_coord = self.precip_cube.coord('realization')
+        threshold_coord.points = [0.5, 7.0, 35.0]
+        threshold_coord.rename('threshold')
+        threshold_coord.units = cf_units.Unit('kg m^-2')
+        self.precip_cube.rename("probability_of_precipitation")
+        self.precip_cube.attributes.update({'relative_to_threshold': 'above'})
+        self.precip_cube.data[1:, 0, ...] = 0.
+        self.precip_cube.coord('forecast_period').points = [4.]
```

Comment:
What are the data values in these cubes?  What size are the arrays?  There is modification but the background initialisation is hidden.  Does `self.fg_cube` have any non-zero values in it?  (Assuming the background value is zero everywhere else?)  This needs to be easy to see without trawling through lines of imported code (which need better docstrings themselves, but that's another conversation!).